php-classes/ActiveRecord.class.php

```diff
@@ -2549,5 +2549,47 @@ protected function getUserFromEnvironment()
         }
 
         return null;
+    },
+
+    /* 
+    *   Handles adding related fields to the query via $dynamicFields.
+    *   Accepts dot-separated string value of relationship (RelationshipName.RelationshipName) or an array of relationships to loop through
+    *   @params ActiveRecord Record
+    *   @params boolean Return only strings
+    *   @params array Options
+    */
+    public function getRelatedData($stringsOnly = false, $options = [])
+    {
+        
+        $relatedObject = null;
+        $value = null;
+        $relationships = $options['relationship'];
+        $Record = $this;
```

Comment:
@nbey don't alias `$this`, keep it clear throughout this function that you're manipulating `$this`